code/modules/vehicles/vehicle_actions.dm

```diff
@@ -306,24 +306,23 @@
 			rider.throw_at(landing_turf, 2, 2)
 			rider.Paralyze(40)
 			vehicle.visible_message(span_danger("[rider] misses the landing and falls on [rider.p_their()] face!"))
-		else
-			rider.spin(4, 1)
-			animate(rider, pixel_y = -6, time = 4)
-			animate(vehicle, pixel_y = -6, time = 3)
-			playsound(vehicle, 'sound/vehicles/skateboard_ollie.ogg', 50, TRUE)
-			passtable_on(rider, VEHICLE_TRAIT)
-			vehicle.pass_flags |= PASSTABLE
-			rider.Move(landing_turf, vehicle_target.dir)
-			passtable_off(rider, VEHICLE_TRAIT)
-			vehicle.pass_flags &= ~PASSTABLE
-		if((locate(/obj/structure/table) in vehicle.loc.contents) || (locate(/obj/structure/fluff/tram_rail) in vehicle.loc.contents))
+		if((locate(/obj/structure/table) in landing_turf) || (locate(/obj/structure/fluff/tram_rail) in landing_turf))
 			if(locate(/obj/structure/fluff/tram_rail) in vehicle.loc.contents)
 				rider.client.give_award(/datum/award/achievement/misc/tram_surfer, rider)
 			vehicle.grinding = TRUE
 			vehicle.icon_state = "[initial(vehicle.icon_state)]-grind"
 			addtimer(CALLBACK(vehicle, /obj/vehicle/ridden/scooter/skateboard/.proc/grind), 2)
 		else
 			vehicle.obj_flags &= ~BLOCK_Z_OUT_DOWN
+		rider.spin(4, 1)
```

Comment:
Doesn't this mean you'll still do the ollie animation if you run out of stanima mid go?